src/Storages/StorageDistributed.cpp

```diff
@@ -1533,10 +1533,7 @@ ClusterPtr StorageDistributed::getOptimizedCluster(
 IColumn::Selector StorageDistributed::createSelector(const ClusterPtr cluster, const ColumnWithTypeAndName & result)
 {
     const auto & slot_to_shard = cluster->getSlotToShard();
-
     const IColumn * column = result.column.get();
-    if (const auto * col_const = typeid_cast<const ColumnConst *>(column))
-        column = &col_const->getDataColumn();
```

Comment:
Function `StorageDistributed::createSelector()` must return a vector specifying how to split the input block into shards according to values of the sharding key. The size of this vector must be the same as the number of rows in the input block, i.e. the number of rows in the `result.column`.

Here this replacement `column = &col_const->getDataColumn()` was wrong because it caused the selector to have only one element. Also this replacement was unnecessary because below in this function the `createBlockSelector()` function is called which knows how to deal with const columns ([see](https://github.com/ClickHouse/ClickHouse/blob/a122857b6eed413e5a9bc985fe0155252847bef6/src/Interpreters/createBlockSelector.cpp#L40-L45)).